Automattic/jetpack
# Protect: Add WAF stats to controls

#### Descriptions
Adds applicable stats to the controls on the `FirewallPage`

#### Changes proposed in this Pull Request:

* Adds conditional stats content to each rules section
* Adds a `setWafStats` action and applies within the `refreshWaf` action to ensure stats are updated when necessary
* Adds the `moment` date library as a Protect dependency

#### Other information:

#### Jetpack product discussion


- 1202232924268038-as-1203286905507147

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:






* Checkout this branch
* Load your Jurassic Tube
* Upgrade to a scan level plan
* Activate Protect and proceed to the Firewall tab
* Enable Automatic rules and verify that the rules version and last update date display as per the screenshots below
* Enable Manual rules to verify that the "No manual rules" notice displays
* Proceed to add rules and ensure that upon returning to the main settings section that the number of rules is correctly reflected in the notice and displays as per the screenshots below
* Add the following the top of the `REST_Controller::api_get_waf()` method and ensure that an installation error is displayed
```
return new WP_Error(
	'error',
	__( 'An error occurred' )
);
```
* Remove the above code and select the toggle once again to ensure that the error notice is appropriately dismissed when the installation is a success
* View each section on mobile to ensure the structure and styling is in line with the provided designs
* Verify that there is no regression in app functionality introduced

#### Screenshots
Stats:
<img width="837" alt="Screen Shot 2022-12-20 at 11 57 00" src="https://user-images.githubusercontent.com/43220201/208755734-cdd7d3e7-5f57-4465-a988-57aca6e88900.png">

No manual rules:
<img width="867" alt="Screen Shot 2022-12-20 at 12 12 28" src="https://user-images.githubusercontent.com/43220201/208757775-dd0db257-8fb8-4a27-b489-add92fdc1794.png">

Installation error:
<img width="885" alt="Screen Shot 2022-12-20 at 12 10 44 (1)" src="https://user-images.githubusercontent.com/43220201/208757542-8e9e3583-508e-444e-9984-06722680d75e.png">
